### PR TITLE
update honeybadger gem

### PR DESCRIPTION
**Pull request by wjordan**
filter honeybadger debug logging and ruby warnings from rake tests

Documentation on Honeybadger upgrade [here](http://docs.honeybadger.io/howto/upgrade_2.html).

---
# Why should I upgrade?

Simply put the new gem is: harder, better, faster, stronger. Not convinced? Then check it:
- It's way faster; all data is now sent asynchronously by default.
- Way better performance: we've tuned object allocation, profiled the memory footprint, and oiled the hinges (except there are no hinges -- just pure elegance). Don't believe us? Under MRI Ruby 2.1, sending 1000 error reports now allocates around **1.2 million** less objects than it used to.
- Better configuration with full support for ENV variables in addition to a new honeybadger.yml file (no more initializers).

---

### CODE REVIEW

**Comment by aoby** [comment]
Make sure the [cronjob](https://github.com/code-dot-org/code-dot-org/blob/bff9c4985d6adf48b1d729cf3ee6517e517f28aa/bin/cronjob#L57-L64) honeybadger functionality is working, as well as dashboard and [pegasus](https://github.com/code-dot-org/code-dot-org/blob/df939f3b8ef110781d8a4d26fdf0acc78ba33507/pegasus/router.rb#L99-L105).

**Comment by wjordan** [comment]
after iterating on an adhoc instance, I updated the pegasus and cronjob configs to work with the new v2 api. thanks andrew for the heads up and pointers on how to test these!

I'll wait for the CI run to pass then merge, and do a final manual test on staging to make sure the integration continues to work there.

File: pegasus/rake/test.rake [hunk 1]
```diff
@@ -1,8 +1,10 @@
 ENV['RACK_ENV'] = 'test'
 
+ENV['HONEYBADGER_LOGGING_LEVEL'] = 'error'
+
 require 'rake/testtask'
 require_relative '../test/fixtures/fake_dashboard.rb'
-Rake::TestTask.new
+Rake::TestTask.new.tap{|x| x.warning = false}
```
Comment (aoby) [inline pegasus/rake/test.rake:7]: Why is this needed?

**Comment by aoby** [comment]
LGTM

(single eyes test failure in circleCI is surely unrelated)

File: pegasus/rake/test.rake (hunk 1 above)
Comment (wjordan) [inline pegasus/rake/test.rake:7]: This is a separate change to filter ruby warnings from rake tests (Rake 11 changed the default for `warnings` from `false` to `true` which started spamming our test runs). I just included this change in this PR since it was low-risk and touched the same file.
